### ISSUE TITLE
'Save as' functionality

### ISSUE THREAD

**Issue by hideheader** [stylish-userstyles/stylish-chrome#65]
There are several elements in this.

1. **Remember if a style has been edited.** Set a 'dirty' flag on the style and/or sections when changes are Saved. Filter styles on this flag in 'Manage'. Probably best to add this information to a new table and `JOIN` it to the existing table(s).

2. **Duplicate an existing style in 'Edit Style'.** If the 'Name' hasn't been changed, replace it with a new name ("copy of ...", <i>i.e.</i>). If the 'Code' has been changed, clear `updateUrl` and `md5Url` (but not `url` - that's its ultimate origin). Optionally, copy the original's `styles.id` into a new field in the new style (that's its immediate origin). Insert all fields (including 'Enabled') into new records, then clear 'Enabled' on the original style and clear the 'dirty' flag in the editor.

3. **Save a copy of a modified style on update or install.** Before overwriting a dirty style, make a disabled copy of it with a new name and no `updateUrl` or `md5Url`. It wouldn't hurt to ask first.

4. **Save a copy of an unmodified style with style settings on install.** Before overwriting a style with style settings, make a disabled copy of it with a new name, retaining  `updateUrl` and `md5Url`. Ask first - sometimes you want two styles with different options, other times you just want to change the options.

5. **Give a new style its own `styles.url`** `styles.id` would suffice.

(1) helps me find styles I've been editing. (2) lets me create an editorial copy while keeping the original. If I haven't made an editorial copy, (3) lets me choose whether to keep my changes or overwrite them. (4) lets me install more than one version of a style with style settings. (5) lets me retrieve versions of styles I've written *de novo*.

(2) also allows me to save successive revisions of a style. Even without recording a copy's provenance you can retrieve all revisions of a style you've installed with 

````SQL
SELECT * FROM styles JOIN ... WHERE styles.url = <url> ORDER BY styles.id DESC
````
However, that won't work with *de novo* styles, so either the immediate source of a copy should be recorded or new styles should be given a unique `styles.url` value.

**Comment by tophf** [comment]
Indeed, duplicating styles is useful for the authors. Currently I have to click to-mozilla button and then my from-mozilla button.
